crates/joycon/src/scan.rs

```diff
@@ -0,0 +1,81 @@
+use crate::JoyCon;
+use anyhow::Error;
+use hidapi::{DeviceInfo, HidApi, HidError};
+use joycon_sys::{
+    input::WhichController, JOYCON_L_BT, JOYCON_R_BT, NINTENDO_VENDOR_ID, PRO_CONTROLLER,
+};
+
+/// Helper for creating [`JoyCon`]s.
+///
+/// The main purpose of this struct is to be returned by [`scan_for_joycons`].
+pub struct DetectedJoyConInfo<'a> {
+    api: &'a HidApi,
+    device_info: &'a DeviceInfo,
+}
+
+/// Helper for [`DetectedJoyConInfo::connect`].
+pub enum ConnectionError {
```

Comment:
This should use thiserror for easier manipulation. BTW I used anyhow in this crate but it really should use thiserror instead.

Reply:
Done.